src/psyclone/tests/psyir/nodes/call_test.py

```diff
@@ -628,6 +629,539 @@ def test_call_get_callees_local(fortran_reader):
+
+    routine_main: Routine = root_node.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    routine_match: Routine = root_node.walk(Routine)[1]
+    assert routine_match.name == "foo"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+    assert call_foo.routine.name == "foo"
+
+    arg_idx_list = []
+
+    try:
+        call_foo.get_callee(ret_arg_match_list=arg_idx_list)
+
+    except CallMatchingArgumentsNotFound as err:
+        assert "No matching routine found for" in str(err)
+        print("Success! Exception triggered (as expected)")
+        return
+
+    assert False, ("This should have triggered an error since there"
+                   "are more arguments in the call than in the routine")
+
+
+def test_call_get_callee_4_named_arguments(fortran_reader):
+    '''
+    Check that named arguments have been correlated correctly
+    '''
+    code = '''
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    call foo(c=e, a=f, b=g)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b, c
+  end subroutine
+
+end module some_mod'''
+
+    root_node: Node = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = root_node.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    routine_match: Routine = root_node.walk(Routine)[1]
+    assert routine_match.name == "foo"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+    assert call_foo.routine.name == "foo"
+
+    arg_idx_list = []
+    result: Routine = call_foo.get_callee(ret_arg_match_list=arg_idx_list)
+
+    assert len(arg_idx_list) == 3
+    assert arg_idx_list[0] == 2
+    assert arg_idx_list[1] == 0
+    assert arg_idx_list[2] == 1
+
+    assert result is routine_match
+
+
+def test_call_get_callee_5_optional_and_named_arguments(fortran_reader):
+    '''
+    Check that optional and named arguments have been correlated correctly
+    '''
+    code = '''
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    call foo(b=e, a=f)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b
+    integer, optional :: c
+  end subroutine
+
+end module some_mod'''
+
+    root_node: Node = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = root_node.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    routine_match: Routine = root_node.walk(Routine)[1]
+    assert routine_match.name == "foo"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+    assert call_foo.routine.name == "foo"
+
+    arg_idx_list = []
+    result: Routine = call_foo.get_callee(ret_arg_match_list=arg_idx_list)
+
+    assert len(arg_idx_list) == 2
+    assert arg_idx_list[0] == 1
+    assert arg_idx_list[1] == 0
+
+    assert result is routine_match
+
+
+def test_call_get_callee_6_interfaces(fortran_reader):
+    '''
+    Check that optional and named arguments have been correlated correctly
```

Comment:
"...correctly when the call is to a generic interface."

Reply:
OK